tensorrt_llm/_torch/custom_ops/cute_dsl_custom_ops.py

```
@@ -8162,6 +8162,211 @@ def _(
+        IndexDecodeScoreKernel
+
+    class CuteDSLMiniMaxM3IndexDecodeScoreRunner:
+        """Runner for the MiniMax-M3 indexer decode block-scoring kernel.
+
+        Caches compiled kernels keyed on the static params
+        (dtype, num_heads, max_decode_query_len, head_dim); batch, query-token
+        count, page count and block-table width stay symbolic, so one compile
+        covers every decode step of a given model shape.
+        """
+
+        kernel_cache = dict()
```

Ruff (0.16.1)

[warning] Mutable default value for class attribute

(RUF012)